gips/0083.md

```diff
@@ -30,6 +30,39 @@ IMPORTANT: Non-normative (or Informative) sections were added for the sake of cl
 
 # Specification
 
+## Permissionless Discovery
+
+This section discusses how Indexers may be listed on the Payments Gateway UI.
+
+Substreams providers MUST register their service on-chain by publishing a _Substreams Service Deployment_ manifest to IPFS, and [`allocate`ing](https://github.com/graphprotocol/contracts/blob/main/packages/contracts/contracts/staking/Staking.sol#L296) to it through the Staking contract on the Arbitrum chain.
+
+Here is a sample _Substreams Service Deployment_ manifest:
+```yaml
+specVersion: 0.0.5
+description: "Substreams Data Service for MyNetwork"
+service:
+  type: substreams-v1
+  endpoint: https://mynetwork.substreams.example.com
+  network: mynetwork
+  provider:
+    id: my-company
+    name: My Company
+    logo: https://company.example.com/logo.png
+```
+
+The service `substreams-v1` means that the endpoint makes available a gRPC endpoint responding to these methods:
+
+- sf.substreams.rpc.v2.Stream/Block
+- sf.substreams.rpc.v2.EndpointInfo/Info
+
+as specified by https://github.com/streamingfast/substreams/blob/develop/proto/sf/substreams/rpc/v2/service.proto updated from time-to-time in backwards compatible ways.
+
+The Payment Gateway MUST listen to the Arbitrum Staking contract for such registrations (through a has a Substreams module streamed, a Subgraph or other means), and update its local view of the network.
+
+The Payment Gateway CAN have systems to perform health checks (e.g. on `/health` or `/info` endpoints), or other checks to ensure active, up-to-date and properly configured providers are offered to end users. A Payment Gateway CAN check block height to ensure the backing provider is close to chain head before offering it to users.
```

Comment:
Future iterations could include public dashboards to help make this system visible. It has some trust assumptions between "Payment Gateway" and Indexers. 
FYI, there are parallel ongoing initiatives related to how Gateways should be monitored, among other things. However, I don't see them as a dependency here, and they also severely slow us down.

Reply:
Good point.